cli/exp_rpty.go

```diff
@@ -210,7 +211,21 @@ func handleRPTY(inv *serpent.Invocation, client *codersdk.Client, args handleRPT
 	_, _ = io.Copy(inv.Stdout, conn)
 	cancel()
 	_ = conn.Close()
-	_, _ = fmt.Fprintf(inv.Stderr, "Connection closed\n")
 
 	return nil
 }
+
+var knownShells = []string{"ash", "bash", "csh", "dash", "fish", "ksh", "powershell", "pwsh", "zsh"}
+
+func isOneShotCommand(cmd []string) bool {
+	// If the command is empty, we'll assume the user wants to open a shell.
+	if len(cmd) == 0 {
+		return false
+	}
+	// If the command is a single word, and that word is a known shell, we'll
+	// assume the user wants to open a shell.
+	if len(cmd) == 1 && slice.Contains(knownShells, cmd[0]) {
+		return false
+	}
+	return true
+}
```

Comment:
This has some interesting repercussions when it's neither a shell and not a one-shot command. Imagine doing `vim /my/file.txt`. Here we would return true but `vim` is one of the reasons we have the screen backend.

This is probably fine though, but I'm flagging this behavior as it's not ideal. At the very least worth a comment.